pkg/controller/volume/persistentvolume/framework_test.go

```diff
@@ -595,22 +599,62 @@ func newTestController(kubeClient clientset.Interface, volumeSource, claimSource
 	if classSource == nil {
 		classSource = fcache.NewFakeControllerSource()
 	}
-
+	// Pass in shared informers as usual but do not start them. Instead,
```

Comment:
It's ugly and in short I have to do this to keep the controller initialization look "normal" at the cost of these unit tests being ugly, but it is needed.

In the unit tests the controller needs to use its own informers built with the the same fake claimsource+volumesource that the volume reactor does. The tests are built around sending things to the volume reactor which does things to pretend to be etcd. So here I pass in shared informers then go ahead and use the special informers instead, which is what the comment is supposed to explain.